Application/src/view/visuals/component/ClassRelationship.java

```diff
@@ -0,0 +1,257 @@
+        // this avoids having 2 loops
+        int stepsParameter;
+
+        /// Using the Bresenham’s Line Algorithm on X axis as reference
+        if (Math.abs(XDistance) > Math.abs(YDistance)) {
+            stepsParameter = Math.abs(XDistance);
+
+            A = 2 * Math.abs(YDistance);
+            B = A - 2 * Math.abs((XDistance));
+            P = A - Math.abs((XDistance));
+            // decreasing the gap to show arrow correctly when it is a horizontal line
+            if (YDistance == 0) {
+                gap += -2;
+            }
+            // decreasing the gap to show arrow correctly when it is a horizontal line pointing right
+            if (XDistance > 0) {
+                gap += -1;
+            }
+            // increasing the gap to show arrow correctly when it is a horizontal line pointing left
+            if (XDistance < 0) {
+                gap += 1;
+            }
+        }
+
+        // Using the Bresenham’s Line Algorithm on the Y axis as reference
+        else {
+            stepsParameter = Math.abs(YDistance);
+
+            A = 2 * Math.abs(XDistance);
+            B = A - 2 * Math.abs((YDistance));
+            P = A - Math.abs((YDistance));
+            // decreasing the gap to show arrow correctly when it is a vertical line line
+            if (XDistance == 0) {
+                gap += -1;
+            }
+            // decreasing the gap to show arrow correctly when it is a vertical line pointing down
+            if (YDistance > 0) {
+                gap += -1;
+            }
+        }
+
+        // This for loop is responsible for drawing the road sprites
+        for (int i = 0; i < stepsParameter / 15 + gap; i++) {
+
+            // if P is less than 0, draw the next sprite on the same line as the last sprite
+            if (P < 0) {
+
+                // increase offset accordingly
+                if (Math.abs(XDistance) > Math.abs(YDistance)) {
+                    XOffset += offsetIncrementX;
+                } else {
+                    YOffset += offsetIncrementY;
+                }
+
+                gc.drawImage(road, startingPointX + XOffset, startingPointY + YOffset, 15, 15);
+
+                P += A; //see Bresenham’s Line Algorithm
+            }
+
+            // if P was 0 or greater, draw the next sprite one line higher/lower than the last sprite
+            if (P >= 0) {
+
+                // increase offset accordingly
+                if (Math.abs(XDistance) > Math.abs(YDistance)) {
+                    XOffset += offsetIncrementX;
+                } else {
+                    YOffset += offsetIncrementY;
+                }
+
+                // increase offset accordingly
+                if (Math.abs(XDistance) > Math.abs(YDistance)) {
+                    YOffset += offsetIncrementY;
+                } else {
+                    XOffset += offsetIncrementX;
+                }
```

Comment:
Line 180-192 I do not see the point of the if statements, you changing the values into the same thing no matter what

Reply:
oh! apparently I had the same if statement twice in a row here!
there should be only 1, and this if statements finds out which offset needs to be incremented, this is better than having 2 separate for loops

Reply:
its not the same if statements, they have the content switched making it so that both the if and the else happens